### PR TITLE
provider/azure: added Azure SQL server and service support.

### PR DESCRIPTION
**Pull request by aznashwan**
Added support, tests and docs for:
* Azure SQL servers
* Azure SQL database services that run on those servers

Any and all reviews are very much appreciated.

### CODE REVIEW

File: builtin/providers/azure/resource_azure_sql_database_service.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		Delete: resourceAzureSqlDatabaseServiceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"collation": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (cbednarski) [inline builtin/providers/azure/resource_azure_sql_database_service.go]: This has to be `Optional: true` since it also has a `Default`.

**Comment by cbednarski** [comment]
LGTM. Will wait for some more eyeballs before merging.

File: builtin/providers/azure/resource_azure_sql_database_service.go (hunk 1 above)
Comment (aznashwan) [inline builtin/providers/azure/resource_azure_sql_database_service.go]: Good catch; also noticed a couple of other discrepancies in the argument definitions and fixed those as well.

**Comment by aznashwan** [comment]
@cbednarski Thanks for the review. :smile:

File: website/source/docs/providers/azure/r/sql_database_server.html.markdown [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    For a list of all Azure locations, please consult [this link](http://azure.microsoft.com/en-us/regions/).
+
+* `username` - (Required) The username for the administrator of the database server.
+
+* `password` - (Required) The password for the administrator of the database server.
+
+* `version` - (Optional) The version of the database server to be used. Can be any
+    one of `2.0` or `12.0`.
+
+* `url` - (Computed) The fully qualified domain name of the databse server.
```
Comment (phinze) [inline website/source/docs/providers/azure/r/sql_database_server.html.markdown]: s/databse/database/

File: website/source/docs/providers/azure/r/sql_database_service.html.markdown [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+* `collation` - (Optional) The collation to be used within the database service.
+    Defaults to the standard Latin charset.
+
+* `max_size_bytes` - (Optional) The maximum size in bytes the database service
+    should be allowed to expand to. Range depends on the database `edition`
+    selected above.
+
+* `service_level_id` - (Optional) The ID corresponding to the service level per
+    edition. Please reffer to [this](https://msdn.microsoft.com/en-us/library/azure/dn505701.aspx) link for more details.
```
Comment (phinze) [inline website/source/docs/providers/azure/r/sql_database_service.html.markdown]: s/reffer/refer/

**Comment by phinze** [comment]
Two wee docs typos, otherwise this is looking great!

**Comment by aznashwan** [comment]
Fixed. Thanks for the review :smile:

**Comment by phinze** [comment]
Once again, great work @aznashwan. Incoming! :rocket:

**Pull request merged**